### PR TITLE
first cut of version on client and server

### PR DESCRIPTION
**Pull request by vaikas**
@adamreese 
@technosophos 
#761
#858

Please take a look and see if you have any grave concerns with this approach :) I need to figure out how to decipher the git tree state, but check the approach please :)

vaikas@vaikas-glaptop:~/projects/go/src/k8s.io/helm$ ./bin/helm --host localhost:44134 version
{Major:2 Minor:0 Patch:0 PreRelease:alpha.3 BuildMetadata:-bf9dea1 GitVersion: GitCommit:bf9dea135199889432226f3a9343049a792dbce0 GitTreeState:}
major:2 pre_release:"alpha.3" build_metadata:"-bf9dea1" git_commit:"bf9dea135199889432226f3a9343049a792dbce0"

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/helm/1126)

### CODE REVIEW

**Comment by vaikas-google** [comment]
@adamreese could you take a looksee please :)

File: pkg/helm/client.go [hunk 1]
```diff
@@ -120,6 +121,20 @@ func (h *Client) UpdateRelease(rlsName string, chStr string, opts ...UpdateOptio
 	return h.opts.rpcUpdateRelease(rlsName, chart, rls.NewReleaseServiceClient(c), opts...)
 }
 
+// Version returns the server version
+//
+// Note: there aren't currently any  supported StatusOptions,
```
Comment (philips) [inline pkg/helm/client.go]: nit: double space

File: pkg/helm/client.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ import (
 
 	"k8s.io/helm/pkg/chartutil"
 	rls "k8s.io/helm/pkg/proto/hapi/services"
+	//	version "k8s.io/helm/pkg/proto/hapi/version"
```
Comment (philips) [inline pkg/helm/client.go]: please don't comment out packages, just remove them

**Comment by philips** [comment]
overall lgtm

File: pkg/helm/client.go (hunk 1 above)
Comment (vaikas-google) [inline pkg/helm/client.go]: done

File: pkg/helm/client.go (hunk 2 above)
Comment (vaikas-google) [inline pkg/helm/client.go]: oops, my bad. removed :)

**Comment by adamreese** [comment]
Is there a reason for adding proto fields for each segment of the semver?  Could we leave as a string and parse if needed?

**Comment by vaikas-google** [comment]
Thanks, can you give the LGTM stamp, or does it need some other work?

**Comment by philips** [comment]
@vaikas-google You still need to answer Adam's question: https://github.com/kubernetes/helm/pull/1126#issuecomment-246008091

**Comment by vaikas-google** [comment]
@adamreese it seemed cleaner to me to break it into separate fields, because they have a separate meaning.Also seems like it would make it less likely for typos, etc. If you feel strongly about it, I can change it back to a string.

**Comment by philips** [comment]
@vaikas-google On the typos thing we could easily just use a semver library and test it in the init function and panic if it is invalid. Semver is a very straightforward spec so I think parsing it at init time as a string is OK.

This seems to be the winning library these days: https://github.com/blang/semver

**Comment by vaikas-google** [comment]
Ok, I'll change it to just a string then.

**Comment by vaikas-google** [comment]
Thanks, PTAL.

**Comment by philips** [comment]
LGTM

**Comment by vaikas-google** [comment]
@adamreese is that what you had in mind, as in good from your side?

File: cmd/helm/version.go [hunk 3]
```diff
@@ -18,20 +18,45 @@ package main
...
+			return version.run()
+		},
+	}
+	return cmd
+}
+
+func (v *versionCmd) run() error {
+	// Regardless of whether we can talk to server or not, just print the client
+	// version.
+	fmt.Printf("%+v\n", version.GetVersionProto())
```
Comment (adamreese) [inline cmd/helm/version.go]: should print to `v.out`

**Comment by adamreese** [comment]
It would help with readability if the formatting was consistent between client and server

```
{SemVer:v2.0.0-alpha.4+4a7352c GitVersion: GitCommit:4a7352cdc06bf22e196e95e84ac99c3a71a1d825 GitTreeState:}
sem_ver:"v2.0.0-alpha.4+4a7352c" git_commit:"4a7352cdc06bf22e196e95e84ac99c3a71a1d825"
```

**Comment by vaikas-google** [comment]
Ok, as per our discussions wrt. release process, I think I addressed everything we talked about and addressed the comments. Here's the current output:
vaikas@vaikas-glaptop:~/projects/go/src/k8s.io/helm$ ./bin/helm --host localhost:44134 version
Client: {SemVer: v2.0.0-alpha.4 GitCommit: 4a7352cdc06bf22e196e95e84ac99c3a71a1d825}
Server: {SemVer: v2.0.0-alpha.4 GitCommit: 4a7352cdc06bf22e196e95e84ac99c3a71a1d825}

**Pull request merged**